### PR TITLE
Add Tunable section

### PR DESCRIPTION
**Pull request by mactanxin**
for ticket https://bugs.freenas.org/issues/15645

### CODE REVIEW

File: ui/inspectors/tunable.reel/tunable.html [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            "properties": {
+                "element": {"#": "inspector"},
+                "canDelete": true,
+                "canSave": true,
+                "canRevert": true
+            },
+            "bindings": {
+                "context": {"<-": "@owner.context"},
+                "isConfirmationVisible": {"<-": "@owner.isConfirmationVisible"},
+                "confirmationMessage": {"<-": "'Are you sure you want to delete ' + @owner.object.id + '?'"}
```
Comment (pchaussalet) [inline ui/inspectors/tunable.reel/tunable.html]: I'm not sure the `object.id` property is the most meaningful to the user to make sure what he is about to delete. I think `object.var` would be a better candidate.

File: ui/inspectors/tunable.reel/tunable.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ */
+exports.Tunable = Component.specialize(/** @lends Tunable# */ {
+    typeOPTIONS: {
+        value: null
+    },
+
+    enterDocument: {
+        value: function(isFirstTime) {
+            var self = this,
+                tunableType = TunableType;
```
Comment (pchaussalet) [inline ui/inspectors/tunable.reel/tunable.js]: Useless, you should just use `TunableType`

File: ui/inspectors/tunable.reel/tunable.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            if (isFirstTime) {
+                this.typeOPTIONS = [];
+                for (var i = 0; i < tunableType.members.length; i++) {
+                    this.typeOPTIONS.push({label: tunableType.members[i], value: tunableType[tunableType.members[i]]});
```
Comment (pchaussalet) [inline ui/inspectors/tunable.reel/tunable.js]: The select-option-converter is made to do that conversion for you.
Just put the content of `TunableType.members` in `this.typeOptions` and instantiate the select-option-converter like this in the serialization: 

```
"tunableTypeOptionConverter": {
    "prototype": "core/converter/select-option-converter",
    "properties": {
        "labelPropertyName": "this",
        "valuePropertyName": "this"
},
```

File: ui/inspectors/tunable.reel/tunable.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+var Component = require("montage/ui/component").Component,
+    Model = require("core/model/model").Model,
+    TunableType = require("core/model/enumerations/tunable-type").TunableType;
+
+/**
+ * @class Tunable
+ * @extends Component
+ */
+exports.Tunable = Component.specialize(/** @lends Tunable# */ {
+    typeOPTIONS: {
```
Comment (pchaussalet) [inline ui/inspectors/tunable.reel/tunable.js]: Please use camelCasing for properties names.

**Review by pchaussalet — CHANGES_REQUESTED** [review]

File: ui/inspectors/tunable.reel/tunable.js [hunk 5]
```diff
@@ -10,18 +10,18 @@ var Component = require("montage/ui/component").Component,
...
-    typeOPTIONS: {
+    typeOptions: {
         value: null
     },
 
     enterDocument: {
         value: function(isFirstTime) {
             var self = this,
-                tunableType = TunableType;
+                EnumTunableType = TunableType;
```
Comment (pchaussalet) [inline ui/inspectors/tunable.reel/tunable.js]: EnumTunableType is useless.

File: ui/inspectors/tunable.reel/tunable.js [hunk 6]
```diff
@@ -10,18 +10,18 @@ var Component = require("montage/ui/component").Component,
...
             if (isFirstTime) {
-                this.typeOPTIONS = [];
-                for (var i = 0; i < tunableType.members.length; i++) {
-                    this.typeOPTIONS.push({label: tunableType.members[i], value: tunableType[tunableType.members[i]]});
+                this.typeOptions = [];
```
Comment (pchaussalet) [inline ui/inspectors/tunable.reel/tunable.js]: You should just assign `TunableType.members` to `this.typeOptions`, no need to iterate by yourself.
Also you should do this in `templateDidLoad` (everything that is run only once and doesn't need to read a bound variable should go in `templateDidLoad`).

File: ui/inspectors/tunable.reel/tunable.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+ * @extends Component
+ */
+exports.Tunable = Component.specialize(/** @lends Tunable# */ {
+    typeOptions: {
+        value: null
+    },
+
+    enterDocument: {
+        value: function(isFirstTime) {
+            var self = this;
```
Comment (pchaussalet) [inline ui/inspectors/tunable.reel/tunable.js]: `self` is not used in this method.

File: ui/inspectors/tunable.reel/tunable.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+            if (isFirstTime) {
+                this.typeOptions = TunableType.members;
```
Comment (pchaussalet) [inline ui/inspectors/tunable.reel/tunable.js]: This should be moved in the `templateDidLoad` method to ensure there is no flickering due to the options loading.

File: ui/inspectors/tunable.reel/tunable.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+            }
+            if (!!this.object._isNew) {
+                if (!this.object.type) {
```
Comment (pchaussalet) [inline ui/inspectors/tunable.reel/tunable.js]: It can be simplified in a single `if` condition.

**Review by pchaussalet — CHANGES_REQUESTED** [review]

File: ui/inspectors/tunable.reel/tunable.js (hunk 7 above)
Comment (mactanxin) [inline ui/inspectors/tunable.reel/tunable.js]: `self` is used

**Review by pchaussalet — APPROVED** [review]

**Pull request merged**